# TOC
- First Steps with Python and Jupyter
- How to run the code
- Performing Arithmetic Operations using Python
- Solving multi-step problems using variables
- Evaluating conditions using Python
- Combining conditions with logical operators
- Adding text styles using Markdown
- Save and upload your notebook
- Further Reading and References
- Questions for Revision
- Solution for Exercise

markdown

# First Steps with Python and Jupyter 

![s](https://i.imgur.com/gvSnw4A.png)



## Performing Arithmetic Operations using Python

Let's begin by using Python as a calculator. You can write and execute Python using a code cell within Jupyter. 

> **Working with cells**: To create a new cell within Jupyter, you can select "Insert > Insert Cell Below" from the menu bar or just press the "+" button on the toolbar. You can also use the keyboard shortcut `Esc+B` to create a new cell. Once a cell is created, click on it to select it. You can then change the cell type to code or markdown (text) using the "Cell > Cell Type" menu option. You can also use the keyboard shortcuts `Esc+Y` and `Esc+M`. Double-click a cell to edit the content within the cell. To apply your changes and run a cell, use the "Cell > Run Cells" menu option or click the "Run" button on the toolbar or just use the keyboard shortcut `Shift+Enter`. You can see a full list of keyboard shortcuts using the "Help > Keyboard Shortcuts" menu option.

Run the code cells below to perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the `//`, `%`, and `**` operators are used for?

In [4]:
2+3+9

14

In [5]:
99-73

26

In [6]:
23.54*-1432

-33709.28

In [7]:
100/7

14.285714285714286

In [8]:
100//7

14

In [9]:
100%7

2

In [10]:
5**3

125

As you might expect, operators like `/` and `*` take precedence over other operators like `+` and `-` as per mathematical conventions. You can use parentheses, i.e. `(` and `)`, to specify the order in which operations are performed.

In [12]:
((2+5) * (17-3)) / (4**3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 

You can use the empty cells below and add more cells if required.

In [14]:
population = 198568
men = 45312
women = 35678
children = population - (men + women)
children

117578

In [16]:
total_pencils = 2425*24
total_pencils

58200

In [17]:
const_of_one = 95844/21
const_of_one

4564.0

In [21]:
annual_production = 24532*12
annual_production

294384

In [22]:
total_bag = 450000
bag_of_sugar = 145968
bag_of_wheat = 236486
bag_of_rice = total_bag - bag_of_sugar - bag_of_wheat
bag_of_rice

67546

In [23]:
product_3th_week = 483685 - 146345 - 138152
product_3th_week

199188

In [25]:
total_cost = 124 * 9372
total_cost

1162128

In [27]:
student_each_room = 4386/86
student_each_room

51.0

In [29]:
no_bus = 1575/75
no_bus

21.0

In [30]:
total_vote = 52496 + 44929 + 36824
total_vote

134249

In [32]:
cost_each_toy = 12960/96
total_amount = 1015 + 12960
cost_each_toy,total_amount

(135.0, 13975)

## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [33]:
500 * (.2 * 1.25)

125.0

Thus, the grocery store makes a total profit of $125. While this is a reasonable way to solve a problem, it's not entirely clear by looking at the code cell what the numbers represent. We can give names to each of the numbers by creating Python *variables*.

> **Variables**: While working with a programming language such as Python, information is stored in *variables*. You can think of variables as containers for storing data. The data stored within a variable is called its *value*.

In [37]:
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

The variables `cost_of_ice_bag`, `profit_margin`, and `number_of_bags` now contain the information provided in the word problem. We can check the value of a variable by typing its name into a cell. We can combine variables using arithmetic operations to create other variables.

> **Code completion**: While typing the name of an existing variable in a code cell within Jupyter, just type the first few characters and press the `Tab` key to autocomplete the variable's name. Try typing `pro` in a code cell below and press `Tab` to autocomplete to `profit_margin`.

In [38]:
profit_per_bag = cost_of_ice_bag * profit_margin
profit_per_bag

0.25

In [39]:
total_profit = number_of_bags * profit_per_bag
total_profit

125.0

If you try to view the value of a variable that has not been *defined*, i.e., given a value using the assignment statement `variable_name = value`, Python shows an error.

In [40]:
net_profit

NameError: name 'net_profit' is not defined

Storing and manipulating data using appropriately named variables is a great way to explain what your code does.

Let's display the result of the word problem using a friendly message. We can do this using the `print` *function*.

> **Functions**: A function is a reusable set of instructions. It takes one or more inputs, performs certain operations, and often returns an output. Python provides many in-built functions like `print` and also allows us to define our own functions.

In [41]:
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


In [42]:
# stor input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# perform the required calcualtions
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


Note that we're using the `#` character to add *comments* within our code. 

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines. 

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [43]:
my_favorite_number = 1 # an inline comment

In [44]:
# this comment gets its own line
my_least_favorite_number = 3

In [45]:
'''
this is a multi-line comment.
Write as little or much as you would like
Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.
Also, if you use good variable names, then your code is
often self explantory, and you may not even need comment
'''
a_neutral_number = 5



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [47]:
total_cost = 5000
people_number = 29
price_per_ticket = 200

total_profit = people_number * price_per_ticket - total_cost

print("the profit made by the company: ",total_profit)

the profit made by the company:  800


## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [49]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [50]:
# Equality check - True
my_favorite_number == 1

True

In [51]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [52]:
# Not equal check-True
my_favorite_number != a_neutral_number

True

In [53]:
# Not equal check-False
a_neutral_number !=3

False

In [55]:
# Greater then check - True
my_least_favorite_number > a_neutral_number

True

In [56]:
# Greater than check - False
my_favorite_number > my_least_favorite_number

False

In [57]:
# Less than check - True
my_favorite_number < 10

True

In [58]:
# Less than  check -False
my_least_favorite_number < my_favorite_number

False

In [59]:
# Greater than or equal check - True
my_favorite_number >=1

True

In [60]:
# Greater than or equal check -False
my_favorite_number >=3

False

In [63]:
# Less than or equal check-True
3+6 <=9

True

In [64]:
# Less than or equal check-False
my_favorite_number + a_neutral_number <=3

False

In [65]:
const_of_ice_bag = 1.25
is_ice_bag_expensive = const_of_ice_bag >=10
print("Is the ice bag expensive? " ,is_ice_bag_expensive)

Is the ice bag expensive?  False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

In [66]:
my_favorite_number

1

In [69]:
my_favorite_number >0 and my_favorite_number <= 3

True

In [70]:
my_favorite_number < 0 and my_favorite_number <=3

False

In [71]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [72]:
True and False

False

In [73]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |




In [77]:
a_neutral_number =3

In [78]:
a_neutral_number ==3 or my_favorite_number < 0

True

In [80]:
a_neutral_number !=3 or my_favorite_number < 0

False

In [81]:
my_favorite_number < 0 or True

True

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [82]:
not a_neutral_number ==3

False

In [83]:
not my_favorite_number < 0

True

In [84]:
not False

True

In [85]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [86]:
(2 > 3 and 4 <= 5) or not (my_favorite_number <0 and True)

True

In [87]:
not (True and 0 <1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [88]:
not True and 0 < 1 or False and True

False

Experiment with arithmetic, conditional and logical operators in Python using the interactive nature of Jupyter notebooks. We will learn more about variables and functions in future tutorials.

There is also a ternary operator in Python. Learn about it here: https://data-flair.training/blogs/python-ternary-operator/

## Adding text styles using Markdown

Adding explanations using text cells (like this one) is a great way to make your notebook informative for other readers. It is also useful if you need to refer back to it in the future. You can double click on a text cell within Jupyter to edit it. In the edit mode, you'll notice that the text looks slightly different (for instance, the heading has a `##` prefix. This text is written using Markdown, a simple way to add styles to your text. Execute this cell to see the output without the special characters. You can switch back and forth between the source and the output to apply a specific style.

For instance, you can use one or more `#` characters at the start of a line to create headers of different sizes:

# Header 1

## Header 2

### Header 3

#### Header 4

To create a bulleted or numbered list, simply start a line with `*` or `1.`.

A bulleted list:

* Item 1
* Item 2
* Item 3

A numbered list:

1. Apple
2. Banana
3. Pineapple

You can make some text bold using `**`, e.g., **some bold text**, or make it italic using `*`, e.g., *some italic text.* You can also create links, e.g., [a link](https://jovian.ai). Images are easily embedded too:

![](https://i.imgur.com/3gjZMYK.png)

Another really nice feature of Markdown is the ability to include blocks of code. Note that code blocks inside Markdown cells cannot be executed.

```
# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

```

You can learn the full syntax of Markdown here: https://learnxinyminutes.com/docs/markdown/